pkg/ovs/ovsconfig/ovs_client.go

```diff
@@ -769,8 +769,12 @@ func (br *OVSBridge) SetInterfaceMTU(name string, MTU int) error {
 	return nil
 }
 
+// GetOVSVersion returns the version of OVS.
 func (br *OVSBridge) GetOVSVersion() (string, Error) {
 	tx := br.ovsdb.Transaction(openvSwitchSchema)
+	if tx == nil {
```

Comment:
This does not look possible: https://github.com/TomCodeLV/OVSDB-golang-lib/blob/9bbdfadcd88184cc76e4e4a589d45cebcf0a66bc/pkg/ovsdb/client.go#L488

Reply:
agree but on our end, with all the layers of packaging and so on, in case this ever fails a fine grained error message will be a lifesaver

Reply:
(i.e. that specific code is in another repo so we cant control for it with unit tests or anything)

Reply:
The function is a constructor and cannot error. If you follow your logic, it's also hard to produce a helpful "fine grained" error message since you do not know what the function is doing. You end up having to look at the code anyway to figure out where the error message comes from and what's happening which IMO is not better than a backtrace. The error message could even be misleading as if there is ever the possibility of the function returning nil, why would it necessary imply that OVS is not installed? And finally, if this constructor is "changed" in the future and errors become possible, I would think that the signature of the function would be updated to return an `error`, which would require us to update the code here anyway.

Reply:
yeah, looking at that code your right all it does is set variables :)

Reply:
ugh.. the fact that it is called `Transaction()` is a big of a naming anti pattern though.  i think ill file an issue on that repo.